src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java

```diff
@@ -312,11 +312,11 @@ public static boolean endsWithChar(String value, char suffix) {
     /**
      * Gets constructor of targetClass.
      *
+     * @param <T> type of the target class object.
```

Comment:
could u please tell me, why is this necessary i mean these issues came with the check so why do we need separate commit for these?

Reply:
This makes the review process easier, since we can review specific commits. If final reviewer wants commits squashed, it is trivial to do before merge.

Reply:
done, total of 7 files in 2nd commit.

Reply:
Not done. Please use the `supplemental` prefix; a good commit message would be something like `supplemental: correct order of param tags in existing Javadocs(#8990)`. This way, your commit is still tied to this issue.

Reply:
done

Reply:
Resolved.